src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/create/actions/ResourcesAllocationAction.java

```diff
@@ -378,12 +378,6 @@ protected void handleError(HaFlowCreateFsm stateMachine, Exception ex, ErrorType
                 listener.onFailed(stateMachine.getHaFlowId(), stateMachine.getErrorReason(), errorType));
     }
 
-    protected HaFlow getHaFlow(String haFlowId) {
-        return haFlowRepository.findById(haFlowId)
-                .orElseThrow(() -> new FlowProcessingException(ErrorType.NOT_FOUND,
-                        format("HA-flow %s not found", haFlowId)));
-    }
-
     private FlowSubType getSubType(int id) {
         //TODO find a better way
```

Comment:
Are you going to find a better way? If no, then I'd create a follow-up task.
The logic probably could be moved to enum itself: have a method FlowSubType.of(int)

Reply:
will create a task